File: apps/web/services/incidents.ts
const BASE_URL = 'http://localhost:8000/api/incidents';

export async function fetchIncidents() {
  const res = await fetch(`${BASE_URL}/`);
  if (!res.ok) throw new Error('Failed to fetch incidents');
  return res.json();
}

export async function fetchServices() {
  const res = await fetch(`${BASE_URL}/services`);
  if (!res.ok) throw new Error('Failed to fetch services');
  return res.json();
}

export async function fetchIncidentById(id: string) {
  const res = await fetch(`${BASE_URL}/${id}`);
  if (!res.ok) throw new Error(`Failed to fetch incident ${id}`);
  return res.json();
}

export async function resolveIncident(id: string) {
  const res = await fetch(`${BASE_URL}/${id}/resolve`, {
    method: 'POST'
  });
  if (!res.ok) throw new Error(`Failed to resolve incident ${id}`);
  return res.json();
}

export async function triggerOutage(service: string, severity: 'critical' | 'warning') {
  const res = await fetch(`${BASE_URL}/trigger-outage?service_name=${encodeURIComponent(service)}&severity=${severity}`, {
    method: 'POST'
  });
  if (!res.ok) throw new Error(`Failed to trigger outage on ${service}`);
  return res.json();
}

File: apps/api/app/api/incidents.py
from fastapi import APIRouter, HTTPException
from app.services.incident_service import incident_service, IncidentUpdate, IncidentCreate

router = APIRouter()

@router.get("/")
def get_all_incidents():
    return incident_service.get_incidents()

@router.get("/services")
def get_services_health():
    return incident_service.get_services()

@router.get("/{incident_id}")
def get_incident_by_id(incident_id: str):
    inc = incident_service.get_incident(incident_id)
    if not inc:
        raise HTTPException(status_code=404, detail="Incident not found")
    return inc

@router.put("/{incident_id}")
def update_incident(incident_id: str, payload: IncidentUpdate):
    inc = incident_service.update_incident_status(incident_id, payload.status)
    if not inc:
        raise HTTPException(status_code=404, detail="Incident not found")
    return inc

@router.post("/{incident_id}/resolve")
def resolve_incident(incident_id: str):
    inc = incident_service.resolve_incident(incident_id)
    if not inc:
        raise HTTPException(status_code=404, detail="Incident not found")
    return inc

@router.post("/trigger-outage")
def trigger_outage(service_name: str, severity: str = "critical"):
    try:
        return incident_service.trigger_service_outage(service_name, severity)
    except ValueError as e:
        raise HTTPException(status_code=400, detail=str(e))
